### PR TITLE
feat: [SRM-15180]: initial implementation of Azure metrics

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
1. Initial implementation of Azure metrics

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts [hunk 1]
```diff
@@ -24,21 +37,50 @@ export function getIsQueryButtonDisabled({
   return Boolean(isEmpty(query) || loading || getIsQueryFieldNotPresent(queryFieldIdentifier, values))
 }
 
+export function getAreAllFieldsNotPopulated(
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts:40]:
```
export function getAreAllFieldsNotPopulated(
  fieldsToFetchRecords: FieldMapping[],
  values: CommonCustomMetricFormikInterface
): boolean {
  const areAllFieldsNotPopulated = fieldsToFetchRecords.some((fieldToFetchRecords) => {
    return !getIsQueryFieldNotPresent(fieldToFetchRecords?.identifier, values);
  });
  
  return !areAllFieldsNotPopulated;
}

```

Comment (deepesh-ui) [reply src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts:40]: Can we try this logic .

File: src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts [hunk 2]
```diff
@@ -24,21 +37,50 @@ export function getIsQueryButtonDisabled({
...
+  queryFieldIdentifier?: string
+  values: CommonCustomMetricFormikInterface
+  query: string
   getString: UseStringsReturn['getString']
-): string {
-  const isQueryFieldNotPresent = getIsQueryFieldNotPresent(queryFieldIdentifier, values)
-  const tooltipMessage = getRunQueryButtonTooltip(query, isQueryFieldNotPresent, queryFieldIdentifier, getString)
+  fieldsToFetchRecords?: FieldMapping[]
+}): string {
+  let tooltipMessage = ''
```
Comment (praneshg239) [inline src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts:74]: we don't need `tooltipMessage` variable. Simply return the string inside the `if` or `else` conditions

File: src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts (hunk 1 above)
Comment (harshilbits) [reply src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts:40]: Tried the above ..one of the testcase breaks .

File: src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts (hunk 2 above)
Comment (harshilbits) [reply src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts:74]: I feel keeping a variable makes it more readable to see how the value changes for different conditions.

**Review by praneshg239 — APPROVED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**